Update `val_labels<-.data.frame` to address issue #192

Theodore-Berkowitz opened the pull request:
Proposed workaround solution for issue #192 where value labels fail to be assigned to completely `NA` columns in a data.frame. Now when an empty column is detected, the function will manually assign the correct mode (from the value labels vector) to the data.frame's column before proceeding with assigning value labels via `val_labels<-`; otherwise, the function will fail with a more informative error message.

The old error message did not report exactly what the problem was (e.g., logical vs. numeric vectors) or what column in the data.frame was causing the issue.

Theodore-Berkowitz committed:
Update val_labels.R (val_labels<-.data.frame)

Handle entirely NA data fields more gracefully. More informative error message.

Theodore-Berkowitz committed:
Update val_labels.R

Added more inline `cli` formatting to error message when data and value modes disagree in `val_labels<-.data.frame`

larmarange commented:
Thanks for the proposal. It may be worth to add a cli message to inform that the type of that variable was converted.

larmarange commented:
By the way, the default method (used for vectors) should be updated as well.

Theodore-Berkowitz commented:
> Thanks for the proposal. It may be worth to add a cli message to inform that the type of that variable was converted.

You're absolutely right! Thank you for suggesting this.



> By the way, the default method (used for vectors) should be updated as well.

Sorry, I'm not familiar with writing methods, so I don't understand what you mean. Are you saying that the function `val_labels<-.default` should also be changed to handle NA vectors the same way I changed the function `val_labels<-.data.frame`? Do we also need to make the same change in `val_labels<-.numeric`, `val_labels<-.character`, and the `haven_labelled` versions?

I've edited the `.default` and `.data.frame` functions as discussed, but please let me know if you think I should change anything else!

Theodore-Berkowitz committed:
Update val_labels.R

Edit default method and data.frame assignment methods to emit warning when coercion happens. Simplified error message in data.frame method for empty (all NA) variables.

larmarange committed:
minors changes + lint

larmarange committed:
NEWS update

fix #192
